src/main/java/io/anserini/server/Controller.java

```diff
@@ -52,4 +59,34 @@ public Map<String, Object> search(@PathVariable(value = "index", required = fals
     return queryMap;
   }
 
+  @RequestMapping(method = RequestMethod.GET, path = "/isCached/{index}")
+  public boolean isCached(@PathVariable("index") String index) {
+    if (!IndexInfo.contains(index)) {
+      throw new IllegalArgumentException("Index name " + index + " not found!");
+    }
+    
+    PrebuiltIndexHandler handler = new PrebuiltIndexHandler(index);
+    handler.initialize();
+    return handler.checkIndexFileExist();
+  }
+
+  @RequestMapping(method = RequestMethod.GET, path = "/list")
```

Comment:
Same comment... what are some alternative route names that might better conform to best practices?